File: TaxDeductionOrganizer/src/TaxDeductionOrganizer.Core/Model/ReceiptAggregate/Receipt.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace TaxDeductionOrganizer.Core;

public class Receipt
{
    public Guid ReceiptId { get; set; }
    public Guid DeductionId { get; set; }
    public string FileName { get; set; } = string.Empty;
    public string FileUrl { get; set; } = string.Empty;
    public DateTime UploadDate { get; set; } = DateTime.UtcNow;
    public string? Notes { get; set; }
    public Deduction? Deduction { get; set; }
}

File: TaxDeductionOrganizer/src/TaxDeductionOrganizer.WebApp/projects/tax-deduction-organizer/src/app/models/receipt.ts
export interface Receipt {
  receiptId: string;
  deductionId: string;
  fileName: string;
  fileUrl: string;
  uploadDate: string;
  notes?: string;
}

export interface CreateReceipt {
  deductionId: string;
  fileName: string;
  fileUrl: string;
  notes?: string;
}

export interface UpdateReceipt {
  receiptId: string;
  deductionId: string;
  fileName: string;
  fileUrl: string;
  notes?: string;
}
